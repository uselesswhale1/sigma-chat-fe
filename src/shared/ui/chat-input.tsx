import {
  InputGroup,
  Input,
  InputRightElement,
  Button,
  Box,
} from "@chakra-ui/react";
import { useState } from "react";
import { PALLETE } from "../constants/color-pallete";

interface ChatInputProps {
  initial?: string;
  onEnter: (input: string) => void;
}

export const ChatInput = ({
  initial = "",
  onEnter,
}: ChatInputProps): JSX.Element => {
  const [message, setMessage] = useState<string>(initial);

  const onApply = () => {
    onEnter(message);
    setMessage("");
  };

  return (
    <Box
      data-testid="chat-test-wrapper"
      p={2}
      bg={PALLETE.bg}
      color={PALLETE.text}
      bottom={0}
      left={0}
      right={0}
      onKeyDown={(e): void => {
        if (e.key === "Enter") {
          e.preventDefault();
          onApply();
        }
      }}
    >
      <InputGroup size="md">
        <Input
          data-testid="chat-test-input"
          pr="4.5rem"
          placeholder="Enter message"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
        />
        <InputRightElement width="4.5rem">
          <Button
            data-testid="chat-enter-button"
            h="1.75rem"
            size="sm"
            onClick={onApply}
            bg={PALLETE.text}
            children="Send"
          />
        </InputRightElement>
      </InputGroup>
    </Box>
  );
};
