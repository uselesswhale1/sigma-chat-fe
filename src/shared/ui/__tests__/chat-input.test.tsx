import { fireEvent, render, screen } from "@testing-library/react";
import { ChatInput } from "../chat-input";
import userEvent from "@testing-library/user-event";

describe("ChatInput should", () => {
  it("match snapshot", () => {
    const enterMock = jest.fn;
    const { container } = render(
      <ChatInput initial="123-321" onEnter={enterMock} />
    );

    expect(container).toMatchSnapshot();
  });

  it("handle text enter", () => {
    const enterMock = jest.fn();

    render(<ChatInput onEnter={enterMock} />);

    const input = screen.getByTestId("chat-test-input");
    const inputBox = screen.getByTestId("chat-test-wrapper");
    const enterBtn = screen.getByTestId("chat-enter-button");

    fireEvent.change(input, { target: { value: "abcc" } });

    fireEvent.keyDown(inputBox, { key: "Enter", code: "Enter", charCode: 13 });

    expect(enterMock).toBeCalledWith("abcc");

    fireEvent.change(input, { target: { value: "ha-haa" } });

    enterBtn.click();

    expect(enterMock).toBeCalledWith("ha-haa");
  });
});
