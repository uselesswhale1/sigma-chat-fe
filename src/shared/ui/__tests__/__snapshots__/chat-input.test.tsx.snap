// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput should match snapshot 1`] = `
<div>
  <div
    class="css-ys80mx"
    data-testid="chat-test-wrapper"
  >
    <div
      class="chakra-input__group css-bx0blc"
      data-group="true"
    >
      <input
        class="chakra-input css-1n0hj84"
        data-testid="chat-test-input"
        placeholder="Enter message"
        value="123-321"
      />
      <div
        class="chakra-input__right-element css-vcnsx7"
      >
        <button
          class="chakra-button css-151vs4d"
          data-testid="chat-enter-button"
          type="button"
        >
          Send
        </button>
      </div>
    </div>
  </div>
</div>
`;
